pkg/handlers/track.go

```diff
@@ -60,13 +60,23 @@ func TrackEvent(w http.ResponseWriter, r *http.Request) {
 		Attributes: body.UserAttributes,
 	}
 
-	err = optlyClient.Track(eventKey, uc, body.EventTags)
-
+	pc, err := optlyClient.ConfigManager.GetConfig()
 	if err != nil {
-		middleware.GetLogger(r).Error().Err(err).Str("eventKey", eventKey).Msg("error tracking event")
+		RenderError(err, http.StatusInternalServerError, w, r)
+		return
+	}
+
+	if _, err = pc.GetEventByKey(eventKey); err != nil {
 		RenderError(err, http.StatusNotFound, w, r)
 		return
 	}
+
+	err = optlyClient.Track(eventKey, uc, body.EventTags)
+	if err != nil {
+		RenderError(err, http.StatusInternalServerError, w, r)
+		return
+	}
+
```

Comment:
Nit: this logic could be inside the `optlyClient`, and it could return well known errors so the caller can decide whether to 500 or 404. Then the handler isn't messing with the config manager or the project config.

Reply:
You're right, this should be part of `OptlyClient`.